server/authz/policy.rego

```diff
@@ -430,9 +430,10 @@ allow {
 # Team admin and maintainers can read and write policies for their teams
 allow {
   not is_null(object.team_id)
-  object.team_id == subject.teams[_].id
+  team_id := subject.teams[_].id
+  object.team_id == team_id
   object.type == "policy"
-  team_role(subject, subject.teams[_].id) == [admin,maintainer][_]
+  team_role(subject, team_id) == [admin,maintainer][_]
```

Comment:
Ah! If we don't assign, we can match to two different teams! Great catch.

Reply:
I wonder if this could be simplified with just `team_role(subject, object.team_id) == [admin,maintainer][_]`

Reply:
Good idea, pushed.